symfony/framework-bundle/4.2/public/index.php

```diff
@@ -7,28 +7,26 @@
 
 require __DIR__.'/../vendor/autoload.php';
 
-// The check is to ensure we don't use .env if APP_ENV is defined
-if (!isset($_SERVER['APP_ENV'])) {
-    if (!class_exists(Dotenv::class)) {
-        throw new \RuntimeException('APP_ENV environment variable is not defined. You need to define environment variables for configuration or add "symfony/dotenv" as a Composer dependency to load variables from a .env file.');
-    }
-    (new Dotenv())->load(__DIR__.'/../.env');
+if (class_exists(Dotenv::class)) {
+    (new Dotenv())->loadForEnv($env ?? 'dev', __DIR__.'/../.env');
```

Comment:
`$env` is never defined here :)

Reply:
same here, if .env loads other env that not dev you will try to load .dev at all times, and it may not be the case / what we want. That would not have happened before.

Reply:
@ogizanagi good catch, fixed.
@joshlopes no, because it uses https://github.com/symfony/symfony/pull/28533